src/Microsoft.ML.Data/Transforms/TransformBase.cs

```diff
@@ -800,13 +809,13 @@ protected override int MapColumnIndex(out bool isSrc, int col)
         protected override IEnumerable<DataViewSchema.Column> GetDependenciesCore(IEnumerable<DataViewSchema.Column> dependingColumns)
             => _bindings.GetDependencies(dependingColumns);
 
-        protected override Delegate[] CreateGetters(DataViewRow input, Func<int, bool> active, out Action disposer)
+        protected override Delegate[] CreateGetters(DataViewRow input, IEnumerable<DataViewSchema.Column> activeColumns, out Action disposer)
         {
             Func<int, bool> activeInfos =
                 iinfo =>
                 {
                     int col = _bindings.MapIinfoToCol(iinfo);
-                    return active(col);
+                    return activeColumns.Select(c => c.Index).Contains(col);
```

Comment:
>activeColumns.Select(c => c.Index) [](start = 27, length = 34)

Make a hashset before you define activeInfos and use hashSet.Contains here.

Reply:
thanks. 

---
In reply to: [261497161](https://github.com/dotnet/machinelearning/pull/2796#discussion_r261497161) [](ancestors = 261497161)